src/syntax/macro.jl

```diff
@@ -1,15 +1,131 @@
-function StenoGraph(ex)
+function StenoGraph_macro(ex)
     if ex.head == :block
         exs = filter(x -> !isa(x, LineNumberNode), ex.args)
         exs = StenoGraphs.variable_as_node!.(exs)
         exs = StenoGraphs.addition_to_vector!.(exs)
         vec = Expr(:call, :vcat, exs...)
-        return esc(:(meld(vcat(StenoGraphs.unarrow.($vec)...))))
+        return esc(:(StenoGraphs.StenoGraph($vec)))
     else
-        StenoGraph(Expr(:block, ex))
+        StenoGraph_macro(Expr(:block, ex))
     end
 end
+"""
+    @StenoGraph
 
+The `@StenoGraph` macro allows you to refer to nodes without explicitly declaring them.
+
+```jldoctest
+@StenoGraph begin
+    # a and b are not declared anywhere but they act as `Node(:a)` & `Node(:b)`
+    a → b
+end
+
+# output
+
+a → b
+```
+
+The main downside is that you can not use regular variables inside the macro.
+Had you declared `c = [Node(:a) Node(:b)]` outside the macro, inside it would nevertheless refer to `Node(:c)`.
+You would need to escape it using [`_`](@ref), i.e. use it as `_(c)` within the macro.
+
+See also [`@declage_nodes`](@ref) & [`@declage_nodes_from`](@ref) for alternatives to escaping.
+
+"""
 macro StenoGraph(ex)
-    StenoGraph(ex)
+    StenoGraph_macro(ex)
+end
+
+"""
+    StenoGraph(x::Vector{ <: Arrow})
+
+Takes a vector of arrows and turns these into a list of edges.
+"""
+StenoGraph(x::Vector{ <: Arrow}) = meld(vcat(unarrow.(x)...))
+StenoGraph(x::Arrow...) = StenoGraph(vcat(x...))
+
+
+function declare_nodes(exs::Symbol...) 
+    out = [] 
+    for var in exs 
+        node = Expr(:call, SimpleNode, QuoteNode(var)) 
+        push!(out, Expr(Symbol("="), var, node)) 
+    end 
+    Expr(:block, out..., nothing) 
+end
+
+"""
+    @declare_nodes(x::Vector{ <: Arrow})
```

Comment:
I don't know how macros are typically documented, but maybe something like this (BenchmarkTools does it similarly):
```suggestion
    @declare_nodes <node names>
```